doc/source/whatsnew/v3.0.0.rst

```diff
@@ -712,6 +712,7 @@ Timezones
 Numeric
 ^^^^^^^
 - Bug in :meth:`DataFrame.corr` where numerical precision errors resulted in correlations above ``1.0`` (:issue:`61120`)
+- Bug in :meth:`DataFrame.cov` where presence of ``NaT`` in columns could lead to incorrect results or errors (:issue:`53115`)
```

Comment:
I think it's worth stating that this raises a `TypeError` now, rather than returning incorrect results. Also IIUC, we *always* raise a `TypeError`, regardless of the presence of `NaT`. So maybe

```suggestion
:meth:`DataFrame.cov` raises a ``TypeError`` instead of returning potentially incorrect results or other errors (:issue:`53115`).
```